scipy/interpolate/_rgi.py

```diff
@@ -217,20 +217,39 @@ def __init__(self, points, values, method="linear", bounds_error=True,
             self._validate_grid_dimensions(points, method)
         self.method = method
         self.bounds_error = bounds_error
+        values = self._check_values(values)
+        self._check_dimensionality(points, values)
+        self._check_fill_value(values, fill_value)
+        self.fill_value = fill_value
```

Comment:
Minor: any reason to not have `self.fill_value = self._check_fill_value(fill_value)`

Reply:
Applied.